src/cli/self_update/windows.rs

```diff
@@ -510,55 +505,49 @@ fn _apply_new_path(new_path: Option<Vec<u16>>) -> Result<()> {
 // Get the windows PATH variable out of the registry as a String. If
 // this returns None then the PATH variable is not a string and we
 // should not mess with it.
-fn get_windows_path_var() -> Result<Option<Vec<u16>>> {
-    use std::io;
-
-    let root = RegKey::predef(HKEY_CURRENT_USER);
-    let environment = root
-        .open_subkey_with_flags("Environment", KEY_READ | KEY_WRITE)
+fn get_windows_path_var() -> Result<Option<HSTRING>> {
+    let environment = CURRENT_USER
+        .create("Environment")
         .context("Failed opening Environment key")?;
 
-    let reg_value = environment.get_raw_value("PATH");
+    let reg_value = environment.get_hstring("PATH");
     match reg_value {
-        Ok(val) => {
-            if let Some(s) = from_winreg_value(&val) {
-                Ok(Some(s))
-            } else {
-                warn!(
-                    "the registry key HKEY_CURRENT_USER\\Environment\\PATH is not a string. \
-                       Not modifying the PATH variable"
-                );
-                Ok(None)
-            }
+        Ok(val) => Ok(Some(val)),
+        Err(e) if e.code() == HRESULT::from_win32(ERROR_INVALID_DATA) => {
+            warn!(
+                "the registry key HKEY_CURRENT_USER\\Environment\\PATH is not a string. \
+                   Not modifying the PATH variable"
+            );
+            Ok(None)
         }
-        Err(ref e) if e.kind() == io::ErrorKind::NotFound => Ok(Some(Vec::new())),
+        Err(e) if e.code() == HRESULT::from_win32(ERROR_FILE_NOT_FOUND) => Ok(Some(HSTRING::new())),
         Err(e) => Err(e).context(CLIError::WindowsUninstallMadness),
     }
 }
 
 // Returns None if the existing old_path does not need changing, otherwise
 // prepends the path_str to old_path, handling empty old_path appropriately.
-fn _add_to_path(old_path: Vec<u16>, path_str: Vec<u16>) -> Option<Vec<u16>> {
+fn _add_to_path(old_path: HSTRING, path_str: HSTRING) -> Option<HSTRING> {
     if old_path.is_empty() {
         Some(path_str)
     } else if old_path
         .windows(path_str.len())
-        .any(|path| path == path_str)
+        .any(|path| path == path_str.deref())
```

Comment:
Calling `deref()` "by hand" is pretty unidiomatic, we'd usually spell it more like `*path_str`.